src/commands/deploy/setPreDeployTaskForDotnet.ts

```diff
@@ -0,0 +1,160 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { MessageItem, TaskDefinition } from 'vscode';
+import { DialogResponses } from 'vscode-azureextensionui';
+import * as constants from '../../constants';
+import { ext } from '../../extensionVariables';
+import { isPathEqual } from '../../utils/pathUtils';
+import { getWorkspaceSetting, updateWorkspaceSetting } from '../../vsCodeConfig/settings';
+import * as tasks from '../../vsCodeConfig/tasks';
+import { IDeployWizardContext } from "./IDeployWizardContext";
+
+export async function setPreDeployTaskForDotnet(context: IDeployWizardContext): Promise<void> {
+    const preDeployTaskSetting: string = 'preDeployTask';
+    const showPreDeployWarningSetting: string = 'showPreDeployWarning';
+    const workspaceFspath: string = context.workspace.uri.fsPath;
+
+    // don't overwrite preDeploy or deploySubpath if it exists and respect configurePreDeployTasks setting
+    if (!getWorkspaceSetting<boolean>(showPreDeployWarningSetting, workspaceFspath)
+        || getWorkspaceSetting<string>(preDeployTaskSetting, workspaceFspath)
+        || getWorkspaceSetting<string>(constants.configurationSettings.deploySubpath, workspaceFspath)) {
+        return;
+    }
+
+    // if the user is deploying a different folder than the root, use this folder without setting up defaults
+    if (!isPathEqual(context.deployFsPath, workspaceFspath)) {
+        return;
+    }
+
+    const csprojFile: string | undefined = await tryGetCsprojFile(context, workspaceFspath);
+
+    // if we found a .csproj file set the tasks and workspace settings
+    if (csprojFile) {
+        const targetFramework: string | undefined = await tryGetTargetFramework(csprojFile);
+        context.telemetry.properties.tfw = targetFramework ? targetFramework : 'N/A';
+        if (!targetFramework) {
+            // if the target framework cannot be found, don't try to set defaults
+            return;
+        }
+
+        const notConfiguredForDeploy: string = `Required assets to build and deploy are missing from "${context.workspace.name}"? Add them?`;
+        const dontShowAgainButton: MessageItem = { title: "No, and don't show again" };
+        const input: MessageItem = await ext.ui.showWarningMessage(notConfiguredForDeploy, { modal: true }, DialogResponses.yes, dontShowAgainButton);
+        if (input === dontShowAgainButton) {
+            await updateWorkspaceSetting(showPreDeployWarningSetting, false, workspaceFspath);
```

Comment:
I thought it'd be better as a workspace setting since this seems like it could be project specific.  I might have a project I know that I never want configured, but maybe other apps I create via `dotnet` it'd be nice for VS Code to handle it.